subiquity/client/controllers/drivers.py

```diff
@@ -0,0 +1,53 @@
+# Copyright 2021 Canonical, Ltd.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import logging
+from typing import List
+
+from subiquitycore.tuicontroller import Skip
+
+from subiquity.common.types import DriversPayload, DriversResponse
+from subiquity.client.controller import SubiquityTuiController
+from subiquity.ui.views.drivers import DriversView
+
+log = logging.getLogger('subiquity.client.controllers.drivers')
+
+
+class DriversController(SubiquityTuiController):
+
+    endpoint_name = 'drivers'
+
+    async def make_ui(self) -> DriversView:
+        response: DriversResponse = await self.endpoint.GET()
+        if not response.drivers and response.drivers is not None:
+            raise Skip
+        return DriversView(self, response.drivers, response.install)
+
+    async def _wait_drivers(self) -> List[str]:
+        response: DriversResponse = await self.endpoint.GET(wait=True)
+        assert response.drivers is not None
+        return response.drivers
+
+    async def run_answers(self) -> None:
+        if 'install' in self.answers:
+            self.done(self.answers["install"])
```

Comment:
I realise this might be my code but in general run_answers should operate the UI a bit more. So this should be something more like self.view.form.install.checked=True; view_helpers.click(self.view.form.ok_btn)

Reply:
Updated. It took me forever to figure out that:
```diff
-click(view.form.done_btn)
+click(view.form.done_btn.base_widget)
```
is necessary so signals get emitted properly.

Also, I didn't keep the text finding stuff because it breaks when the content is translated. Instead I stored a status variable in the view.

Reply:
Oh right yes, I guess we should keep this translation neutral!